src/core/transaction.c

```diff
@@ -1159,6 +1159,9 @@ static bool shall_stop_on_isolate(Transaction *tr, Unit *u) {
         if (u->ignore_on_isolate)
                 return false;
 
+        if (u->survive_soft_reboot)
+                return false;
```

Comment:
Why?

Reply:
same reason IgnoreOnIsolate has to be set manually, otherwise it gets stopped